File: kbignum-java/src/test/java/BigDecimalTest.java
import com.papacekb.kbignum.BigDecimal;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class BigDecimalTest {

    @Test
    public void testBigDecimal() {

        BigDecimal a = new BigDecimal(42.0);
        BigDecimal b = new BigDecimal(21.0);

        BigDecimal expected = new BigDecimal(63.0);
        BigDecimal actual = a.add(b);

        assertEquals(expected, actual);
        System.out.println(expected);
        System.out.println(actual);
    }
}

File: kbignum-mobile/kbignum/src/commonTest/kotlin/com.papacekb.kbignum/BigDecimalTest.kt
package com.papacekb.kbignum

import kotlin.test.Test
import kotlin.test.assertEquals

class BigDecimalTest {

    @Test
    fun testBigDecimalAddition() {
        val a = BigDecimal(42.0)
        val b = BigDecimal(21.0)
        assertEquals(BigDecimal(63.0), a.add(b))
    }

    @Test
    fun testBigDecimalSubtraction() {
        val a = BigDecimal(42.5)
        val b = BigDecimal(33.1)
        assertEquals(BigDecimal(9.4), a.subtract(b))
    }

    @Test
    fun testBigDecimalMultiplication() {
        val a = BigDecimal(42)
        val b = BigDecimal(1.4)
        assertEquals(BigDecimal(58.8), a.multiply(b))
    }

    @Test
    fun testBigDecimalDivision() {
        val a = BigDecimal(42.0)
        val b = BigDecimal(6.4)
        assertEquals(BigDecimal(6.5625), a.divide(b))
    }
}